sdk1/src/test/java/com/amazonaws/services/dynamodbv2/mapper/integration/MissingSignatureFieldsITCase.java

```diff
@@ -0,0 +1,168 @@
+    // Insert the data
+    dynamo.putItem(new PutItemRequest(TABLE_NAME, attrsWithNoEncryptionFlags));
+
+    attrsWithEncryptionFlags =
+        encryptor.encryptAllFieldsExcept(attrsWithEncryptionFlags, context, KEY_NAME);
+    dynamo.putItem(new PutItemRequest(TABLE_NAME, attrsWithEncryptionFlags));
+
+    attrsWithMissingSignatureEncryptionFlag =
+        encryptor.encryptAllFieldsExcept(
+            attrsWithMissingSignatureEncryptionFlag, context, KEY_NAME);
+    attrsWithMissingSignatureEncryptionFlag.remove(encryptor.getSignatureFieldName());
+    dynamo.putItem(new PutItemRequest(TABLE_NAME, attrsWithMissingSignatureEncryptionFlag));
+
+    attrsWithMissingMaterialDescEncryptionFlag =
+        encryptor.encryptAllFieldsExcept(
+            attrsWithMissingMaterialDescEncryptionFlag, context, KEY_NAME);
+    attrsWithMissingSignatureEncryptionFlag.remove(encryptor.getMaterialDescriptionFieldName());
+    dynamo.putItem(new PutItemRequest(TABLE_NAME, attrsWithMissingMaterialDescEncryptionFlag));
+  }
+
+  @Test
+  public void testLoadWithMissingSignatureAndMaterialDescFields() {
+    DynamoDBMapper util = TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo);
+    AllDoNotTouchTable load =
+        util.load(AllDoNotTouchTable.class, attrsWithNoEncryptionFlags.get(KEY_NAME).getS());
+
+    assertEquals(load.getKey(), attrsWithNoEncryptionFlags.get(KEY_NAME).getS());
+    assertEquals(
+        load.getStringAttribute(), attrsWithNoEncryptionFlags.get(STRING_ATTRIBUTE).getS());
+  }
+
+  @Test(
+      expectedExceptions = DynamoDBMappingException.class,
+      expectedExceptionsMessageRegExp = "java.security.SignatureException: Bad signature")
+  public void testLoadWithBadMissingSignatureField() {
+    TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo)
+        .load(EncryptedTable.class, attrsWithMissingSignatureEncryptionFlag.get(KEY_NAME).getS());
+  }
+
+  @Test(
+      expectedExceptions = DynamoDBMappingException.class,
+      expectedExceptionsMessageRegExp = "java.security.SignatureException: Bad signature")
+  public void testLoadWithBadMissingMaterialDescField() {
+    TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo)
+        .load(
+            EncryptedTable.class, attrsWithMissingMaterialDescEncryptionFlag.get(KEY_NAME).getS());
+  }
+
+  @Test(
+      expectedExceptions = DynamoDBMappingException.class,
+      expectedExceptionsMessageRegExp =
+          "java.lang.IllegalArgumentException: Record did not contain encryption metadata fields: '\\*amzn-ddb-map-sig\\*', '\\*amzn-ddb-map-desc\\*'.")
+  public void testLoadWithBadMissingSignatureNMaterialDescFields() {
+    TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo)
+        .load(EncryptedTable.class, attrsWithNoEncryptionFlags.get(KEY_NAME).getS());
+  }
+
+  @DynamoDBTable(tableName = "aws-java-sdk-util-crypto")
+  public static class AllDoNotTouchTable {
```

Comment:
Since we are going to be needing at least three different modeled objects for our above tests, instead of defining our own, let's reuse some which have already been defined:
- [BaseClass](https://github.com/aws/aws-dynamodb-encryption-java/blob/master/sdk1/src/test/java/com/amazonaws/services/dynamodbv2/testing/types/BaseClass.java) can be used where we are currently using `EncryptedTable`
- [Untouched](https://github.com/aws/aws-dynamodb-encryption-java/blob/master/sdk1/src/test/java/com/amazonaws/services/dynamodbv2/testing/types/Untouched.java) can be used where we are currently using `AllDoNotTouchTable`
- [UntouchedWIthNewAttribute](https://github.com/aws/aws-dynamodb-encryption-java/blob/master/sdk1/src/test/java/com/amazonaws/services/dynamodbv2/testing/types/UntouchedWithNewAttribute.java) can be used for the new test I want to add.

Sorry that this is slightly different feedback than I gave you last time, but the modeled objects I link above are fairly well defined and unexpected to change, and re-using them will make these tests a bit nicer.

Reply:
The idea is good but there seems to be an implementation restriction, i.e. this class extends [DynamoDBCryptoIntegrationTestBase](https://github.com/aws/aws-dynamodb-encryption-java/blob/873b7a725f4df7c0fea0a337fffea8ba82bb717f/sdk1/src/test/java/com/amazonaws/services/dynamodbv2/mapper/integration/DynamoDBCryptoIntegrationTestBase.java#L37) where all the table init logic is present and this table init logic is restricted to use `KEY_NAME = "key"`

So using the classes that you mentioned may not be possible. Instead added 
`UntouchedTable` , `UntouchedWithNewEncryptedAttributeTable` and `EncryptedTable` model in this class itself, which is giving better readability as well.

Reply:
I see. In that case, what is currently here is fine :+1: